polyfills/fm/client/fmradio.js

```diff
@@ -0,0 +1,421 @@
+      // Antenna available should be updated
+      if (typeof this.realAntennaAvailable !== 'undefined' &&
+          this.realAntennaAvailable !== null) {
+            return this.realAntennaAvailable;
+      }
+
+      _createAndQueueRequest({
+        operation: 'get',
+        name: 'antennaAvailable'
+      });
+      return this.fakeAntennaAvailable;
+    },
+
+    get frequency() {
+      // Current frequency value should be updated
+      if (typeof this.realFrequency !== 'undefined' &&
+        this.realFrequency !== null) {
+          return this.realFrequency;
+      }
+
+      _createAndQueueRequest({
+        operation: 'get',
+        name: 'frequency'
+      });
+      // Let's assume that we're in the lower bound frequency
+      // This is a trick beacuse the current frequency could be null if the
+      // radio is not enabled and we need to draw the UI with an initial
+      // frequency
+      this.realFrequency = this.fakeFrequencyLowerBound;
+
+      return this.fakeFrequencyLowerBound;
+    },
+
+    get frequencyUpperBound() {
+      if (typeof this.realFrequencyUpperBound !== 'undefined' &&
+        this.realFrequencyUpperBound !== null) {
+          return this.realFrequencyUpperBound;
+      }
+
+      _createAndQueueRequest({
+        operation: 'get',
+        name: 'frequencyUpperBound'
+      });
+      return this.fakeFrequencyUpperBound;
+    },
+
+    get frequencyLowerBound() {
+      if (typeof this.realFrequencyLowerBound !== 'undefined' &&
+        this.realFrequencyLowerBound !== null) {
+          return this.realFrequencyLowerBound;
+      }
+
+      _createAndQueueRequest({
+        operation: 'get',
+        name: 'frequencyLowerBound'
+      });
+      return this.fakeFrequencyLowerBound;
+    },
+
+    get channelWidth() {
+      if (typeof this.realChannelWidth !== 'undefined' &&
+        this.realChannelWidth !== null) {
+          return this.realChannelWidth;
+      }
+
+      _createAndQueueRequest({
+        operation: 'get',
+        name: 'channelWidth'
+      });
+      return this.fakeChannelWidth;
+    },
+
+    _onchange: function(changeType, property, cb) {
+      this['_on' + changeType] = cb;
+      this['_on' + changeType + 'Id'] = this['_on' + changeType + 'Id'] ||
+        ++_currentRequestId;
+      var self = this;
+      var commandObject = {
+        serialize: function() {
+          return {
+            id: ++_currentRequestId,
+            data: {
+              operation: 'onpropertychange',
+              handler: 'on' + changeType,
+              property: property
+            },
+            processAnswer: answer => {
+              if (answer.result) {
+                self[_getPropertyName(property)] = answer.result.propertyValue;
+                cb(answer.result);
+              }
+            }
+          };
+        }
+      };
+      navConnPromise.then(navConnHelper =>
+        navConnHelper.sendObject(commandObject));
+    },
+
+    set onfrequencychange(cb) {
```

Comment:
NIT: you can do all of those 'on' change on a loop... something like
['onev1', 'onev2'].forEach(eventType => 
   Object.defineProperty(this, eventType, {whatever}));

It's not really needed for you to change this now, but it will make for less clutter :).